src/common/utils/tar.ts

```diff
@@ -0,0 +1,55 @@
+// Helper for working with tarball files (.tar, .tgz)
+// Docs: https://github.com/npm/node-tar
+import tar, { ExtractOptions, FileStat } from "tar";
+import path from "path";
+
+export interface ReadFileFromTarOpts {
+  tarPath: string;
+  filePath: string;
+  parseJson?: boolean;
+}
+
+export function readFileFromTar<R = Buffer>({tarPath, filePath, parseJson}: ReadFileFromTarOpts): Promise<R> {
+  return new Promise(async (resolve, reject) => {
+    const fileChunks: Buffer[] = [];
+
+    await tar.list({
+      file: tarPath,
+      filter: path => path === filePath,
+      onentry(entry: FileStat) {
+        entry.on("data", chunk => {
+          fileChunks.push(chunk);
+        });
+        entry.once("error", err => {
+          reject(`Reading ${entry.path} error: ${err}`);
```

Comment:
Why not?

Reply:
For a number of reasons, you loose the locality of the error (where it came from). It is no longer possible to get the source error (because it has been stringified). We should be able to inspect errors. What I really would like to see is something like the following:

```ts
export class ReadTarError {
    constructor(public path: string, public source: Error) {}

    toString() {
        return `Reading ${this.path} failed. ${this.source}`;
    }
}
```